functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,637 @@
+				},
+				"spec": map[string]interface{}{
+					"match": map[string]interface{}{
+						"scope":      "Namespaced",
+						"namespaces": []string{includedNamepsace.GetName()},
+					},
+					"location": "metadata.labels.ns-included",
+					"parameters": map[string]interface{}{
+						"assign": map[string]interface{}{
+							"value": "true",
+						},
+					},
+				},
+			},
+		}
+		_, err = client.Resource("assignmetadata").Create(context.Background(), namespaceScopeIncluded, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), namespaceScopeIncluded.GetName(), metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		clusterScopeSelected := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "mutations.gatekeeper.sh/v1alpha1",
+				"kind":       "AssignMetadata",
+				"metadata": map[string]interface{}{
+					"name": "cluster-scope-selected",
+				},
+				"spec": map[string]interface{}{
+					"match": map[string]interface{}{
+						"scope": "Cluster",
+						"labelSelector": map[string]interface{}{
+							"matchLabels": map[string]interface{}{
+								"selected": "true",
+							},
+						},
+					},
+					"location": "metadata.labels.cluster-selected",
+					"parameters": map[string]interface{}{
+						"assign": map[string]interface{}{
+							"value": "true",
+						},
+					},
+				},
+			},
+		}
+		_, err = client.Resource("assignmetadata").Create(context.Background(), clusterScopeSelected, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), clusterScopeSelected.GetName(), metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}()
+
+		namespaceSelected := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "mutations.gatekeeper.sh/v1alpha1",
+				"kind":       "AssignMetadata",
+				"metadata": map[string]interface{}{
+					"name": "namespace-selected",
+				},
+				"spec": map[string]interface{}{
+					"match": map[string]interface{}{
+						"scope": "Namespaced",
+						"namespaceSelector": map[string]interface{}{
+							"matchLabels": map[string]interface{}{
+								"ns-selected": "true",
+							},
+						},
+						"excludedNamespaces": []string{excludedNamespace.GetName()},
+					},
+					"location": "metadata.labels.ns-selected",
+					"parameters": map[string]interface{}{
+						"assign": map[string]interface{}{
+							"value": "true",
+						},
+					},
+				},
+			},
+		}
+		_, err = client.Resource("assignmetadata").Create(context.Background(), namespaceSelected, metav1.CreateOptions{})
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			err := client.Resource("assignmetadata").Delete(context.Background(), namespaceSelected.GetName(), metav1.DeleteOptions{})
+			Expect(err).NotTo(HaveOccurred())
+		}()
```

Comment:
It should be more readable if we defined the Assigns in one place and then created them in a loop. Something like:

```
allScopeSelected = {...}
namespaceScopeIncluded = {...}
namespaceScopeIncluded= {...}
...

mutations = { allScopeSelected,  namespaceScopeIncluded, namespaceScopeIncluded}

for mutation ;= range mutatinos {
    client.Resource("assignmetadata").Create(context.Background(), namespaceSelected, metav1.CreateOptions{})
    defer func(){}
}
```